src/addons/blog/services/blog.ts

```diff
@@ -91,10 +98,34 @@ export class AddonBlogProvider {
      * @returns Entry id.
      * @since 4.4
      */
-    async addEntry(params: AddonBlogAddEntryWSParams, siteId?: string): Promise<number> {
+    async addEntry({ created, ...params }: AddonBlogAddEntryWSParams & { created: number }, siteId?: string): Promise<void> {
         const site = await CoreSites.getSite(siteId);
 
-        return await site.write<number>('core_blog_add_entry', params);
+        const storeOffline = async (): Promise<void> => {
+            await AddonBlogOffline.addOfflineEntry({
+                ...params,
+                userid: site.getUserId(),
+                lastmodified: created,
+                options: JSON.stringify(params.options),
+                created,
+            });
+        };
+
+        if (!CoreNetwork.isOnline()) {
+            return await storeOffline();
+        }
+
+        try {
+            return await site.write('core_blog_add_entry', params);
+        } catch (error) {
+            if (error && !CoreUtils.isWebServiceError(error)) {
```

Comment:
```suggestion
            if (!CoreUtils.isWebServiceError(error)) {
```
I think there's no need to check if `error` is set, `isWebServiceError` is already checking it. And if `error` is empty we should probably consider it is as a local error, which `isWebServiceError` already does.

The same happens in the `updateEntry` and `deleteEntry` functions.